src/aind_codeocean_utils/codeocean_job.py

```diff
@@ -0,0 +1,582 @@
+"""Module with generic Code Ocean job"""
+import logging
+import time
+from datetime import datetime
+from typing import Dict, List, Optional, Tuple, Union
+
+import requests
+from aind_codeocean_api.codeocean import CodeOceanClient
+from aind_codeocean_api.models.computations_requests import RunCapsuleRequest
+from aind_codeocean_api.models.data_assets_requests import (
+    CreateDataAssetRequest,
+    Source,
+    Sources,
+)
+
+LOG_FMT = "%(asctime)s %(message)s"
+LOG_DATE_FMT = "%Y-%m-%d %H:%M"
+
+logging.basicConfig(format=LOG_FMT, datefmt=LOG_DATE_FMT)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class CodeOceanJob:
+    """
+    This class contains convenient methods to register data assets,
+    run capsules, and capture results.
+    """
+
+    default_datetime_format = "%Y-%m-%d_%H-%M-%S"
+
+    def __init__(self, co_client: CodeOceanClient):
```

Comment:
It will probably be better to define a Settings object which the user can set the Job definitions in and then pass it into this class.

Reply:
This class can do several things: 
- register an asset
- run a capsule from an existing asset
- run a capsule after creating an asset
- capture a result

I think it would be better to pass specific settings to the single functions instead

Reply:
I think though, the code ocean client already has an api to do each of these functions. The CapsuleJob was supposed to help orchestrate running those methods in sequence and passing information from one step to the other. If the user is required to maintain the reference to "ecephys_642478_2023-01-09_13-00-27" (for example) in several places, it might be prone to errors vs. keeping information that in a class attribute. I like Configs classes since it makes it easier to parse yaml, json, command line arguments, etc., but I'd be okay if we just kept information in class attributes.

Reply:
Good point! Do you have time to give it a stab?

Reply:
I think I can take a look at this over the weekend. Is that time frame okay?

Reply:
No rush, we can also finish it up together first week of January :)